core/services/relay/evm/functions.go

```diff
@@ -82,3 +90,73 @@ func newFunctionsConfigProvider(pluginType functionsRelay.FunctionsPluginType, c
 
 	return newConfigWatcher(lggr, contractAddress, contractABI, offchainConfigDigester, cp, chain, relayConfig.FromBlock, args.New), nil
 }
+
+func newFunctionsContractTransmitter(contractVersion uint32, rargs relaytypes.RelayArgs, transmitterID string, configWatcher *configWatcher, ethKeystore keystore.Eth, lggr logger.Logger) (ContractTransmitter, error) {
+	var relayConfig evmRelayTypes.RelayConfig
+	if err := json.Unmarshal(rargs.RelayConfig, &relayConfig); err != nil {
+		return nil, err
+	}
+	var fromAddresses []common.Address
+	sendingKeys := relayConfig.SendingKeys
+	if !relayConfig.EffectiveTransmitterID.Valid {
+		return nil, errors.New("EffectiveTransmitterID must be specified")
+	}
+	effectiveTransmitterAddress := common.HexToAddress(relayConfig.EffectiveTransmitterID.String)
+
+	sendingKeysLength := len(sendingKeys)
+	if sendingKeysLength == 0 {
+		return nil, errors.New("no sending keys provided")
+	}
+
+	// If we are using multiple sending keys, then a forwarder is needed to rotate transmissions.
+	// Ensure that this forwarder is not set to a local sending key, and ensure our sending keys are enabled.
+	for _, s := range sendingKeys {
+		if sendingKeysLength > 1 && s == effectiveTransmitterAddress.String() {
```

Comment:
Not sure I understand this.  I assume this is copy-pasted from elsewhere?

Reply:
Yeah, copy-pasted but I can still explain. It's for a forwarder (a separate node that is responsible for transmission - some products use them). I don't think we need that and I almost removed it but decided to keep for now. I'll ask Foundations about the plans for forwarders and then decide.